File: apps/api/FinanceControl.Shared/Dtos/Response/LegalDocumentResponseDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Response
{
    public class LegalDocumentResponseDto
    {
        public EnumLegalDocumentType Type { get; set; }
        public int Version { get; set; }

        /// <summary>Markdown, exactly as published.</summary>
        public string Content { get; set; } = string.Empty;

        public string ContentHash { get; set; } = string.Empty;
        public DateTime PublishedAt { get; set; }
    }
}

File: apps/mobile/lib/features/legal/data/dtos/legal_document_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'legal_document_response_dto.g.dart';

@JsonSerializable(createToJson: false)
class LegalDocumentResponseDto {
  const LegalDocumentResponseDto({
    required this.type,
    required this.version,
    required this.content,
    required this.contentHash,
    required this.publishedAt,
  });

  final String type;
  final int version;

  /// Markdown, exactly as published.
  final String content;

  final String contentHash;
  final DateTime publishedAt;

  factory LegalDocumentResponseDto.fromJson(Map<String, dynamic> json) =>
      _$LegalDocumentResponseDtoFromJson(json);
}
